src/bos/common/clients/bos/components.py

```diff
@@ -23,19 +23,31 @@
 #
 import logging
 
-from .base import BaseBosEndpoint
+from .base import BaseBosNonTenantAwareEndpoint
+from .options import options
 
 LOGGER = logging.getLogger(__name__)
 
 
-class ComponentEndpoint(BaseBosEndpoint):
+class ComponentEndpoint(BaseBosNonTenantAwareEndpoint):
     ENDPOINT = __name__.lower().rsplit('.', maxsplit=1)[-1]
 
     def get_component(self, component_id):
         return self.get_item(component_id)
 
     def get_components(self, **kwargs):
-        return self.get_items(**kwargs)
+        page_size = options.max_component_batch_size
+        if page_size == 0:
+            return self.get_items(**kwargs)
+        next_page = self.get_items(page_size=page_size, **kwargs)
+        results = next_page
+        while len(next_page) == page_size:
+            last_id = next_page[-1]["id"]
+            next_page = self.get_items(page_size=page_size,
+                                       start_after_id=last_id,
+                                       **kwargs)
+            results.extend(next_page)
+        return results
```

Comment:
What is return data looks like at line#50? an example will help.

Reply:
This is another case where the original function didn't have that info, so I didn't add it. All of the existing BOS client methods are that way. While it is not as newbie friendly, in this case it is probably because they pretty much directly map to BOS API calls, so it would only be really important to spell that out if what was being returned was something different than what the API call produces. In this case, the get_components function returns a list of components (as dicts), just like the API call.

One thing that will also help with this is a separate PR I have underway where I add type annotations to the places where they are lacking. For example, for this function, it will have the return type as something like `list[ComponentData]`, and `ComponentData` will probably be a `TypedDict` that you can examine to see the exact structure.

But as previously discussed, I don't want to use this PR to overhaul existing code more than it already is, due to its already large scope.